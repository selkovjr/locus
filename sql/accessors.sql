--
--  Locus datatype test
--
-- Testng accessors
--
SELECT contig('7:10000-20000'::locus);
SELECT contig('chr7:10000-20000'::locus);
SELECT lower('7:10000-20000'::locus);
SELECT upper('7:10000-20000'::locus);
SELECT length('7:10000-20000'::locus);
SELECT range('7:10000-20000'::locus);
SELECT center('7:10000-20000'::locus);

SELECT lower('7:10000'::locus);
SELECT upper('7:10000'::locus);
SELECT length('7:10000'::locus);
SELECT range('7:10000'::locus);
SELECT center('7:10000'::locus);

SELECT lower('7'::locus);
SELECT upper('7'::locus);
SELECT length('7'::locus);
SELECT range('7'::locus);
SELECT center('7'::locus);
