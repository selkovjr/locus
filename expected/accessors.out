--
--  Locus datatype test
--
-- Testng accessors
--
SELECT contig('7:10000-20000'::locus);
 contig
--------
 7
(1 row)

SELECT contig('chr7:10000-20000'::locus);
 contig
--------
 7
(1 row)

SELECT lower('7:10000-20000'::locus);
 lower
-------
 10000
(1 row)

SELECT upper('7:10000-20000'::locus);
 upper
-------
 20000
(1 row)

SELECT length('7:10000-20000'::locus);
 length
--------
  10000
(1 row)

SELECT range('7:10000-20000'::locus);
     range
---------------
 [10000,20000]
(1 row)

SELECT center('7:10000-20000'::locus);
 center
--------
  15000
(1 row)

SELECT lower('7:10000'::locus);
 lower
-------
 10000
(1 row)

SELECT upper('7:10000'::locus);
 upper
-------
 10000
(1 row)

SELECT length('7:10000'::locus);
 length
--------
      0
(1 row)

SELECT range('7:10000'::locus);
     range
---------------
 [10000,10000]
(1 row)

SELECT center('7:10000'::locus);
 center
--------
  10000
(1 row)

SELECT lower('7'::locus);
 lower
-------
     0
(1 row)

SELECT upper('7'::locus);
   upper
------------
 2147483647
(1 row)

SELECT length('7'::locus);
   length
------------
 2147483647
(1 row)

SELECT range('7'::locus);
     range
----------------
 [0,2147483647]
(1 row)

SELECT center('7'::locus);
   center   
------------
 1073741823
(1 row)

